C/plugins/north/OMF/plugin.cpp

```diff
@@ -1553,42 +1548,145 @@ int PIWebAPIGetVersion(CONNECTOR_INFO* connInfo, bool logMessage)
 	_PIWebAPI->setAuthMethod          (connInfo->PIWebAPIAuthMethod);
 	_PIWebAPI->setAuthBasicCredentials(connInfo->PIWebAPICredentials);
 
-	int httpCode = _PIWebAPI->GetVersion(connInfo->hostAndPort, connInfo->PIWebAPIVersion, logMessage);
+	int httpCode = _PIWebAPI->GetVersion(connInfo->hostAndPort, connInfo->RestServerVersion, logMessage);
 	delete _PIWebAPI;
 
 	return httpCode;
 }
 
+static std::string ParseEDSProductInformation(std::string json)
+{
+	std::string version;
+
+	Document doc;
+
+	if (!doc.Parse(json.c_str()).HasParseError())
+	{
+		try
+		{
+			if (doc.HasMember("Edge Data Store"))
+			{
+				const rapidjson::Value &EDS = doc["Edge Data Store"];
+				version = EDS.GetString();
+			}
+		}
+		catch (...)
+		{
+		}
+	}
+
+	Logger::getLogger()->debug("Edge Data Store Version: %s JSON: %s", version.c_str(), json.c_str());
+	return version;
+}
+
+int EDSGetVersion(CONNECTOR_INFO *connInfo)
```

Comment:
We should have doxygen comment blocks on new functions.

Reply:
Done. Also added a new routine _ParseProductVersion_ so that the major/minor version parsing are not done directly in _SetOMFVersion_.